### PR TITLE
feat: amcl work areas MAP-55

### PR DESCRIPTION
**Pull request by 27Apoorva**

### CODE REVIEW

**Review by ccpowers — APPROVED** [review]

File: amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -959,6 +1009,9 @@ AmclNode::handleMapMessage(const nav_msgs::OccupancyGrid& msg)
   // try to apply the initial pose now that the map has arrived.
   applyInitialPose();
```
Comment (schayapathy) [inline amcl/src/amcl_node.cpp:1010]: In apply_initial_pose function particle filter is initialized using  initial_poses_hyp and in updatePoseFromServer function inital_pose variable is updated.   
I feel we need to update initial_poses_hyp as well in order to apply the pose correctly.

pf_init(pf_, initial_poses_hyp_.size(), &initial_means[0], &initial_covs[0]);

File: amcl/src/amcl_node.cpp [hunk 2]
```diff
@@ -794,8 +812,35 @@ void AmclNode::savePoseToServer()
...
+  {
+    ROS_INFO("Using pose from server since we have all params.");
+    return true;
+  }
+  return false;
+}
+
 void AmclNode::updatePoseFromServer()
 {
+  // default values
```
Comment (schayapathy) [inline amcl/src/amcl_node.cpp:843]: Can we reuse checkPoseParam on sever in this function and reduce duplicate checks.

Comment (27Apoorva) [reply amcl/src/amcl_node.cpp:843]: so I did this to maintain code re-usability. There is a way especially in tests where you can just mention few pose params like initial_pose_x and intial_pose_y and for rest of params, use default. That is what `updatePoseFromServer()` does. But we set our own covariance for apriltags and we want ALL params to be set by us. That is why I added this `checkPoseParam` for getting information back to core. Does that justify two functions?

File: amcl/src/amcl_node.cpp (hunk 1 above)
Comment (27Apoorva) [reply amcl/src/amcl_node.cpp:1010]: we don't need to initialize initial_poses_hyp_ as it  applyInitialPose clears it up and will just exit

**Pull request merged**